Automattic/wp-calypso
# Jetpack Backup: Update multisites to allow backup archive download

#### Changes proposed in this Pull Request

For multisite sites that already have a Backup subscription (or Plan that includes Backup) from prior to when we began restricting the purchase of Backup for Multisites, this PR makes the Backup status screen available so the user can download Backup archives.  One-click restores are removed (because restores are not supported with multisite).

Originated from discussion: pbuNQi-1jg-p2
Related to: 6356-gh-Automattic/jpop-issues

Fixes # 1164141197617539-as-1200312739878522

Implementation notes:
- For multisite installs that have a Backup subscription, we remove the Backup status card that says, "WordPress multi-sites are not supported."  (See image)
    - <img width="764" alt="Screenshot on 2021-05-24 at 09-36-01" src="https://user-images.githubusercontent.com/11078128/119355769-a62a8f00-bc73-11eb-9016-624c2c58fada.png">
- Instead, we go ahead and show the standard Backup Status card with date picker.
- But add a message that says, "Your site, is a WordPress Multisite installation. Jetpack Backup for Multisite installations provides downloadable backups, no one-click restores. For more information visit our documentation page."
- We allow backup downloads (show "Download backup" buttons.
- But remove "Restore to this point" buttons.


![Markup 2021-05-25 at 16 55 17](https://user-images.githubusercontent.com/11078128/119567297-26d3b300-bd7a-11eb-8c7e-0fd134be0c79.png)



#### Testing instructions

1. Checkout and run this PR in both Calypso blue and green concurrently.
2. Create a JN Multisite install and connect Jetpack. Just Jetpack "Free", for now.
3. In Calypso, select Jetpack --> Backup.    ( `http://calypso.localhost:3000/backup/:site` ) and verify you see the "WordPress multi-sites are not supported." backup status card.
4. Using the Store Admin, add a Jetpack Backup subscription to the site (or any Plan that includes Backup).  Backup Realtime product or plan is preferred because then you'll be able to view the "More backups from this day" ActivityCard component containing the additional Download/Restore action buttons.
5. Now that the site has an active Backup subscription, In Calypso select Jetpack --> Backup.    ( `http://calypso.localhost:3000/backup/:site` )
6. Verify you see the Backup Status page like the screenshot above.
7. Verify the Multisite notice/message shows and looks like the screenshot.
8. Verify the "Download backup" button is showing and looks like screenshot.
9. Verify the "Restore to this point" button does not show. (as shown in screenshot)
10. Select a backup and click to "Download" the backup. Verify you are able to Download backups.
11. Verify this is all the same on Calypso Green also.

#### Unit test

This is for the `siteHasSubscription( state, siteId, subscriptionSlug )` selector.

run `yarn test-client client/state/selectors/test/site-has-subscription.js`

1. Verify all tests pass.
